vertical-pod-autoscaler/pkg/updater/logic/updater.go

```diff
@@ -51,32 +53,53 @@ type Updater interface {
-	evictionRateLimiter     *rate.Limiter
-	selectorFetcher         target.VpaTargetSelectorFetcher
+	vpaLister                    vpa_lister.VerticalPodAutoscalerLister
+	podLister                    v1lister.PodLister
+	eventRecorder                record.EventRecorder
+	evictionFactory              eviction.PodsEvictionRestrictionFactory
+	recommendationProcessor      vpa_api_util.RecommendationProcessor
+	evictionAdmission            priority.PodEvictionAdmission
+	evictionRateLimiter          *rate.Limiter
+	selectorFetcher              target.VpaTargetSelectorFetcher
+	useAdmissionControllerStatus bool
+	statusClient                 *status.Client
 }
 
 // NewUpdater creates Updater with given configuration
-func NewUpdater(kubeClient kube_client.Interface, vpaClient *vpa_clientset.Clientset, minReplicasForEvicition int, evictionRateLimit float64, evictionRateBurst int, evictionToleranceFraction float64, recommendationProcessor vpa_api_util.RecommendationProcessor, evictionAdmission priority.PodEvictionAdmission, selectorFetcher target.VpaTargetSelectorFetcher) (Updater, error) {
+func NewUpdater(
+	kubeClient kube_client.Interface,
+	vpaClient *vpa_clientset.Clientset,
+	minReplicasForEvicition int,
+	evictionRateLimit float64,
+	evictionRateBurst int,
+	evictionToleranceFraction float64,
+	useAdmissionControllerStatus bool,
+	recommendationProcessor vpa_api_util.RecommendationProcessor,
+	evictionAdmission priority.PodEvictionAdmission,
+	selectorFetcher target.VpaTargetSelectorFetcher,
+) (Updater, error) {
 	evictionRateLimiter := getRateLimiter(evictionRateLimit, evictionRateBurst)
 	factory, err := eviction.NewPodsEvictionRestrictionFactory(kubeClient, minReplicasForEvicition, evictionToleranceFraction)
 	if err != nil {
 		return nil, fmt.Errorf("Failed to create eviction restriction factory: %v", err)
 	}
 	return &updater{
-		vpaLister:               vpa_api_util.NewAllVpasLister(vpaClient, make(chan struct{})),
-		podLister:               newPodLister(kubeClient),
-		eventRecorder:           newEventRecorder(kubeClient),
-		evictionFactory:         factory,
-		recommendationProcessor: recommendationProcessor,
-		evictionRateLimiter:     evictionRateLimiter,
-		evictionAdmission:       evictionAdmission,
-		selectorFetcher:         selectorFetcher,
+		vpaLister:                    vpa_api_util.NewAllVpasLister(vpaClient, make(chan struct{})),
+		podLister:                    newPodLister(kubeClient),
+		eventRecorder:                newEventRecorder(kubeClient),
+		evictionFactory:              factory,
+		recommendationProcessor:      recommendationProcessor,
+		evictionRateLimiter:          evictionRateLimiter,
+		evictionAdmission:            evictionAdmission,
+		selectorFetcher:              selectorFetcher,
+		useAdmissionControllerStatus: useAdmissionControllerStatus,
+		statusClient: status.NewClient(
+			kubeClient,
+			amission_controller_logic.VpaAdmissionControllerStatusName,
```

Comment:
It seems weird that updater would import admission_controller_logic. Perhaps we can move the consts to the status lib which is common?

Reply:
Good idea. I created https://github.com/kubernetes/autoscaler/pull/2785 to do this.